browser/js/common/factories/user.factory.js

```diff
@@ -0,0 +1,47 @@
+app.factory('UserFactory', function($http) {
+
+    var successFunc = function(res) { return res.data};
+
+    return {
+
+        fetchUser: function(userId) {
+            return $http.get('/api/users/' + userId)
+                .then(successFunc)
+                .catch();
+        },
+
+        addDataset: function(user, dataset, domain) {
+            if (domain) {
+                return $http.post('/api/users/' + user.id + '/addSocrataDataset', { dataset, domain })
```

Comment:
I think I remember Omri saying something about not having verbs in URI's in general. Maybe make this to user.id/SocrataDateset and since it's a POST we'll know we're adding?